pymode/libs/logilab-common-1.4.1/logilab/common/modutils.py

```diff
@@ -671,24 +676,27 @@ def _module_file(modpath, path=None):
         # __path__, get back information from there
         module = sys.modules[modpath.pop(0)]
         # use list() to protect against _NamespacePath instance we get with python 3, which
-        # find_module later doesn't like
+        # find_spec later doesn't like
         path = list(module.__path__)
         if not modpath:
             return C_BUILTIN, None
     imported = []
     while modpath:
         modname = modpath[0]
-        # take care to changes in find_module implementation wrt builtin modules
+        # take care to changes in find_module/find_spec implementation wrt builtin modules
         #
         # Python 2.6.6 (r266:84292, Sep 11 2012, 08:34:23)
-        # >>> imp.find_module('posix')
+        # >>> importlib.find_module('posix')
         # (None, 'posix', ('', '', 6))
         #
         # Python 3.3.1 (default, Apr 26 2013, 12:08:46)
-        # >>> imp.find_module('posix')
+        # >>> importlib.find_module('posix')
+        # (None, None, ('', '', 6))
+        # Python 3.40 (default, Apr 26 2013, 12:08:46)
+        # >>> importlib.find_spec('posix')
         # (None, None, ('', '', 6))
         try:
-            _, mp_filename, mp_desc = find_module(modname, path)
+            _, mp_filename, mp_desc = find_spec(modname, path)
```

Comment:
```TypeError: cannot unpack non-iterable ModuleSpec object```

Reply:
`find_module` and `find_spec` are not a replacement for each other, they work in completely different ways.